### ISSUE TITLE
[BUG] op_math adding two GellMann observables doesn't treat them as different Hermitians

### ISSUE THREAD

**Issue by Gabriel-Bottrill** [PennyLaneAI/pennylane#4341]
### Expected behavior

```
>>> gell_1 = qml.GellMann(wires=0, index=3)
>>> gell_2 = qml.GellMann(wires=0, index=8)
>>> ham_added = gell_1 + gell_2
>>> print(type(ham_added))
<class 'pennylane.ops.qubit.hamiltonian.Hamiltonian'>
>>> print(ham_added.coeffs)
array([1,1])
>>> print(ham_added.ops)
[GellMann3(wires=[0]), GellMann8(wires=[0])]
```

The two GellMann observables should be appended in a Hamiltonian object as shown.

### Actual behavior

```
>>> gell_1 = qml.GellMann(wires=0, index=3)
>>> gell_2 = qml.GellMann(wires=0, index=8)
>>> ham_added = gell_1 + gell_2
>>> print(type(ham_added))
<class 'pennylane.ops.qubit.hamiltonian.Hamiltonian'>
>>> print(ham_added.coeffs)
array([2])
>>> print(ham_added.ops)
[GellMann3(wires=[0])]
```
Currently when adding two GellMann observables Pennylane interprets them as the same type and concatinates them together taking the first as it's type.

### Additional information

By using ```qml.operation.enable_new_opmath()``` the bahaviour seems to work correctly.

```
>>> qml.operation.enable_new_opmath()

>>> gell_1 = qml.GellMann(wires=0, index=3)
>>> gell_2 = qml.GellMann(wires=0, index=8)
>>> ham_added = gell_1 + gell_2
>>> print(type(ham_added))
<class 'pennylane.ops.op_math.sum.Sum'>
>>> print(ham_added)
GellMann3(wires=[0]) + GellMann8(wires=[0])
```

Similar to #3114, on comment chain I was told to make a new issue

### System information

```shell
Name: PennyLane
Version: 0.32.0.dev0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/PennyLaneAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /usr/local/lib/python3.10/dist-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, rustworkx, scipy, semantic-version, toml
Required-by: PennyLane-Lightning

Platform info:           Linux-5.15.107+-x86_64-with-glibc2.31
Python version:          3.10.12
Numpy version:           1.22.4
Scipy version:           1.10.0
Installed devices:
- default.gaussian (PennyLane-0.32.0.dev0)
- default.mixed (PennyLane-0.32.0.dev0)
- default.qubit (PennyLane-0.32.0.dev0)
- default.qubit.autograd (PennyLane-0.32.0.dev0)
- default.qubit.jax (PennyLane-0.32.0.dev0)
- default.qubit.tf (PennyLane-0.32.0.dev0)
- default.qubit.torch (PennyLane-0.32.0.dev0)
- default.qutrit (PennyLane-0.32.0.dev0)
- null.qubit (PennyLane-0.32.0.dev0)
- lightning.qubit (PennyLane-Lightning-0.31.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by albi3ro** [comment]
Thanks for opening this issue @Gabriel-Bottrill .

We could potentially override the dunder methods on qutrit observables so they always use operator arithmetic.
